docs/design/communication-protocol/publish-block-stream.md

```diff
@@ -72,113 +77,248 @@ stream and retry (either to another Block Node, or after a short delay).
 
-    Note over Publisher,BlockNode: Repeat process until block number matches
+This extension adds to the base protocol to describe how the Publisher and
+Block-Node will interact when there are multiple Publishers connected to a
+single Block-Node.
+
+Nothing in this extension changes the base protocol. In particular the
+verification and acknowledgement processes are unchanged.
+
+The core element of multiple-publisher support is that only one Publisher
+should send the full block content for a given block, but this should also
+be the "fastest" Publisher for each block. The Block-Node will accept the first
+Block Header received for each block, and ask all the other publishers to not
+send the remainder of the block; resuming with the next block.
+![Conceptual Diagram](../../assets/Multi-Publisher-Diagram.svg)
+If any situation arises where the received block data is incomplete, not valid,
+or becomes slow or delayed, the Block-Node will ask all publishers to resend
+that block and again select the "fastest", with the exception that the previous
+"winner" cannot be selected for the failed block. This ensures that a publisher
+that is "unhealthy" will not be reselected for the same failed block.
+
+* Publisher, on connect, sends a block header, this contains a block number.
+  * If this publisher is the _first_ to send that block header, then processing
+    continues as in the base protocol.
+  * If this publisher is _not_ the first to send that block header, then
+    the Block-Node will reply with "SkipBlock".
+    * The Publisher will not send the remainder of that block, but will resume
+      sending data with the next Block Header.
+      * This Publisher _may_ be able to resume _before_ the `BlockProof` for
+        the current block is available, and _should_ do so.
+    * The Publisher _must_ still expect acknowledgement for the "skipped" block
+      and _must not_ remove that block from cache until it is acknowledged.
+    * A Publisher _may_ receive a "ResendBlock" response from the Block-Node
+      if the Publisher must resend the "skipped" block. The Published should
+      then resume sending data from the "skipped" block Header for the block
+      number specified in `ResendBlock`.
+      * Note, it is entirely possible to receive another "SkipBlock" after
+        receiving a "ResendBlock" response, and should be handled as above.
+        This is possible because a Block-Node will send the "ResendBlock" to
+        all Publishers at the same time, and another Publisher may still
+        "win the race" to send the full block.
+  * If, while receiving a block from one node, another node _completes_ a later
+    block, the Block-Node _may_ choose to send `EndStream` with status `TIMEOUT`
+    to the current streaming Publisher for the earlier block, and _must_ then
+    send `ResendBlock` to all other publishers, specifying the block number that
+    _was_ being sent by the first Publisher.
+  * Example
```

Comment:
I like this example as an illustration. I think we need more of them to highlight the appropriateness of many of the codes